core/lib/dal/src/contract_verification_dal.rs

```diff
@@ -181,238 +172,219 @@ impl ContractVerificationDal<'_, '_> {
-                WHERE id = $1
-                ",
-                id as i64,
-                error.as_str(),
-                &compilation_errors,
-                panic_message
-            )
-            .execute(self.storage.conn())
-            .await?;
-            Ok(())
-        }
+    ) -> sqlx::Result<()> {
+        sqlx::query!(
+            "
+            UPDATE contract_verification_requests
+            SET status = 'failed', updated_at = now(), error = $2, compilation_errors = $3, panic_message = $4
+            WHERE id = $1
+            ",
+            id as i64,
+            error.as_str(),
+            &compilation_errors,
+            panic_message
+        )
+        .execute(self.storage.conn())
+        .await?;
+        Ok(())
     }
 
     pub async fn get_verification_request_status(
         &mut self,
         id: usize,
-    ) -> Result<Option<VerificationRequestStatus>, SqlxError> {
-        {
-            let result = sqlx::query!(
-                "
-                SELECT status, error, compilation_errors FROM contract_verification_requests
-                WHERE id = $1
-                ",
-                id as i64,
-            )
-            .fetch_optional(self.storage.conn())
-            .await?
-            .map(|row| VerificationRequestStatus {
-                status: row.status,
-                error: row.error,
-                compilation_errors: row
-                    .compilation_errors
-                    .and_then(|errors: serde_json::Value| {
-                        let string_array: Vec<String> = errors
-                            .as_array()
-                            .unwrap()
-                            .iter()
-                            .map(|value| value.as_str().unwrap().to_string())
-                            .collect();
-                        if string_array.is_empty() {
-                            None
-                        } else {
-                            Some(string_array)
-                        }
-                    }),
-            });
-            Ok(result)
-        }
+    ) -> sqlx::Result<Option<VerificationRequestStatus>> {
+        let result = sqlx::query!(
+            "
+            SELECT status, error, compilation_errors FROM contract_verification_requests
+            WHERE id = $1
+            ",
+            id as i64,
+        )
+        .fetch_optional(self.storage.conn())
+        .await?
+        .map(|row| VerificationRequestStatus {
+            status: row.status,
+            error: row.error,
+            compilation_errors: row
+                .compilation_errors
+                .and_then(|errors: serde_json::Value| {
+                    let string_array: Vec<String> = errors
+                        .as_array()
+                        .unwrap()
+                        .iter()
+                        .map(|value| value.as_str().unwrap().to_string())
```

Comment:
Just to clarify: Do we consider it appropriate to `unwrap()` here and in similar places because we assume that the JSON shape will never get corrupted (e.g., by manually inserting a non-conforming value to the table)?

Reply:
nah, I somehow missed these unwraps

Reply:
fixed